Formula/n/neomutt.rb

```diff
@@ -22,6 +22,7 @@ class Neomutt < Formula
   depends_on "tcl-tk" => :build
   depends_on "gettext"
   depends_on "gpgme"
+  depends_on "libiconv"
```

Comment:
This needs to be macOS-only.

Reply:
How about?

```ruby
on_macos do
  # macOS Sonoma ships with a broken libiconv
  depends_on "libiconv"
end
``` 

@roccoblues 
Unfortunately, you made your own fork, so I can update the PR.

Reply:
I think we also need to guard the configure option. My plan is to update this PR when the new version is ready.